source/opt/instruction.cpp

```diff
@@ -439,8 +453,19 @@ bool Instruction::IsReadOnlyVariableShaders() const {
   return is_nonwritable;
```

Comment:
When this was suppose to be for variables only, this check would be appropriate because the decoration is on the variable.  However, will there be a need for this to be expanded?  For example, if you ask if the result of an OpAccessChain is read-only, should it recursively check if the input pointer is read-only?

You don't need to implement it.  This function was only intended to return turn if is definitely know that it is read only.  However, returning false does not guarantee that it is not read-only.  How please leave a comment if you are not going to implement it.

Reply:
Good point!

I'll leave a comment to this effect, because the general solution could be fairly complex (for example, knowing whether a function parameter is definitely read only would I suppose require analysing all calls to the function).

Reply:
Done.